### PR TITLE
Handle shareable levels when previewing them as a teacher

### PR DESCRIPTION
**Pull request by sderickson**
## When hovering over the "Game" link at the top.
screenshot 2016-09-28 13 18 35

## After finishing the level
screenshot 2016-09-28 13 18 31

### CODE REVIEW

File: app/models/CourseInstance.coffee [hunk 1]
```diff
@@ -21,11 +21,12 @@ module.exports = class CourseInstance extends CocoModel
       data: { userID: userID }
     }
     _.extend options, opts
-    @fetch options
+    jqhr = @fetch options
```
Comment (Imperadeiro98) [inline app/models/CourseInstance.coffee]: Should't it be `jqxhr`?

Comment (sderickson) [inline app/models/CourseInstance.coffee]: Ah! Good catch. Going to put some tests on these methods...

File: app/templates/play/level/tome/spell-top-bar-view.jade [hunk 2]
```diff
@@ -26,12 +26,23 @@
...
+        data-trigger="hover"
+      )
+    else
+      - var url = '/play/' + view.options.level.get('type') + '-level/' + view.options.level.get('slug') + '/' + view.options.session.id;
+      - if (view.options.courseID) url += '?course=' + view.options.courseID;
+      a.btn.btn-small.btn-illustrated(href=url)
+        if view.options.level.isType('game-dev')
+          span(data-i18n='sharing.game')
+        else
+          span(data-i18n='sharing.webpage')
```
Comment (Imperadeiro98) [inline app/templates/play/level/tome/spell-top-bar-view.jade]: These 4 lines should make use of `i18nName` in line 29?

Comment (sderickson) [inline app/templates/play/level/tome/spell-top-bar-view.jade]: That was part of logic I removed, decided not to refactor that bit. Removed that vestigial piece of code.

Comment (Imperadeiro98) [inline app/templates/play/level/tome/spell-top-bar-view.jade]: But `i18nName` is being used on line 31

File: app/locale/en.coffee [hunk 3]
```diff
@@ -1527,6 +1527,8 @@
   sharing:
     game: "Game"
     webpage: "Webpage"
+    your_students_preview: "Your students click here to preview their creation as their classmates would see it! Unavailable in teacher preview."
```
Comment (differentmatt) [inline app/locale/en.coffee]: `Your students will click here to see their finished projects! Unavailable in teacher preview.`

File: app/templates/play/level/tome/spell-top-bar-view.jade (hunk 2 above)
Comment (sderickson) [inline app/templates/play/level/tome/spell-top-bar-view.jade]: I... must not have gotten enough sleep yesterday. Fixed this, and verified everything still works. Thanks for tracking this! Here's [the latest version](https://github.com/codecombat/codecombat/pull/3951/files#diff-120a447993ea84be9cab9eb97510fa86R29).

File: app/locale/en.coffee (hunk 3 above)
Comment (sderickson) [inline app/locale/en.coffee]: Copy updated. Agreed a preview of a generic solution would be ideal, but that can be set up later, separately.

**Pull request merged**